app/modules/topics/components/Editor/index.js

```diff
@@ -47,41 +69,78 @@ class PureEditor extends React.Component<Props> {
     onSave();
   };
 
+  componentWillUnmount = (): void => {
+    const { topic, discard } = this.props;
+
+    if (topic.isDirty) discard(topic.id);
+  };
+
   fetchCondition = (topic: ?m.Topic): boolean => {
     return (topic == null || !topic.isContentFetched);
   };
 
   renderEditor = (topic: m.Topic): React.Node => {
-    const { t } = this.props;
+    const { t, setDirty, beforeUnloadHandler } = this.props;
+
+    // Prompt when user refreshes window with unsaved changes
+    if (topic.isDirty) {
+      window.addEventListener('beforeunload', beforeUnloadHandler);
+    }
+    else {
+      window.removeEventListener('beforeunload', beforeUnloadHandler);
+    }
 
     return (
       <div data-test-id="topic-editor">
-        <div style={{ overflow: 'hidden' }}>
-          <Header floated="left" as="h1">{topic.title}</Header>
-          <Button
-            floated="right"
-            primary={true}
-            icon={true}
-            labelPosition="left"
-            onClick={this.handleSaveButtonClick}
-            data-test-id="topic-editor-save-button"
-          >
-            <Icon name="save" />
-            {t('common:button.save')}
-          </Button>
-          <Button
-            floated="right"
-            icon={true}
-            labelPosition="left"
-            data-test-id="topic-editor-cancel-button"
-            as={Link}
-            to={USER_PROFILE_ROUTE}
-          >
-            <Icon name="cancel" />
-            {t('common:button.cancel')}
-          </Button>
-        </div>
-        <ContentItemEditableDisplay contentItemId={topic.rootContentItemId} />
+        {/* Prompt when user navigates away from the page with unsaved changes */}
+        {/* TODO: actually discard the changes when the user wants to leave */}
+        <Prompt
+          when={topic.isDirty}
+          message={t('topics:modals.unsavedChanges.message')}
+        />
+
+        <Grid>
```

Comment:
Note: the reason I didn't use a grid for this in the first place is that it breaks the layout on small page widths / long topic titles. Allthough there are already so many issues with responsiveness that one more or less doesn't make that much of a difference, I guess 😅